processing/src/main/java/io/druid/segment/IndexMerger.java

```diff
@@ -549,10 +553,10 @@ private File makeIndexFiles(
     IOPeon ioPeon = new TmpFileIOPeon();
     ArrayList<FileOutputSupplier> dimOuts = Lists.newArrayListWithCapacity(mergedDimensions.size());
     Map<String, Integer> dimensionCardinalities = Maps.newHashMap();
-    ArrayList<Map<String, IntBuffer>> dimConversions = Lists.newArrayListWithCapacity(indexes.size());
+    ArrayList<Map<String, int[]>> dimConversions = Lists.newArrayListWithCapacity(indexes.size());
```

Comment:
previously the IntBuffer was allocated off-heap, with this change it will be on-heap. Whether this is better or worse can be debated. Currently we rely on garbage collection to hopefully cleanup our buffers before we run out of memory, but allocating it on-heap may create lots of heap pressure if those arrays are big and long-lived. It might be worth some benchmarks to get a sense of how things will be affected.